server/sonar-server/src/main/java/org/sonar/server/rule/index/RuleIndexDefinition.java

```diff
@@ -106,6 +107,7 @@ public void define(IndexDefinitionContext context) {
     ruleMapping.stringFieldBuilder(FIELD_RULE_TEMPLATE_KEY).docValues().build();
 
     ruleMapping.stringFieldBuilder(FIELD_RULE_ALL_TAGS).enableGramSearch().build();
+    ruleMapping.stringFieldBuilder(FIELD_RULE_TYPE).enableGramSearch().build();
```

Comment:
No need for gram search on type, for example searching type "VULN" instead of "VULNERABILITY" is not useful.